### DISCUSSION TITLE
How to replace the value of http request parameter

### DISCUSSION THREAD

**Discussion by M00nBack** [PortSwigger/BChecks#73] [General]
I just found request.url.query, but didn't find a way to replace request parameters. I think it is important, for example, to detect log4j vulnerabilities in passive scanning.

**Comment by Hannah-PortSwigger** [comment]
Hi.

Could you clarify the specific parameters that you are interested in replacing?

If you're using an insertion-point BCheck, then any parameters will be used as an insertion point already.

Additionally, you can always use "regex_replace" to find and replace specific content in your request.

**Reply by M00nBack** [reply]
Thanks for answering!

Yes, insertion-point can indeed be used, but what I want is to be able to judge the value of the parameter before replacing it. I saw a similar problem at: https://github.com/PortSwigger/BChecks/issues/13

Using regex_replace is quite cumbersome.

**Reply by Hannah-PortSwigger** [reply]
Unfortunately, you're not currently able to retrieve the value of an insertion point.

We have been gathering feedback for improvements that we can make to BChecks, and this is a common request that people have been making. We've added your +1 to this feature request.

In the meantime, you may need to use regex_replace as a workaround.

If you're looking for Log4Shell specifically, then you could check out the "Log4Shell Everywhere" extension on the BApp Store, or the ["Log4Shell Scanner" extension](https://github.com/PortSwigger/log4shell-scanner) that was previously on the BApp Store. Unfortunately, this extension had to be removed from the BApp Store as the Log4Shell payload was triggering some anti-virus tooling.

**Reply by M00nBack** [reply]
There is another question, how to contribute the BChecks code? I only found a packaged jar, is BChecks ready to open source?

**Reply by Hannah-PortSwigger** [reply]
You can contribute BChecks to this repository to share your written checks with others.

Check out our contribution guide [here](https://github.com/PortSwigger/BChecks/blob/main/CONTRIBUTING.md)!

It's not possible to modify the BCheck language - this is written within Burp's code, which is closed-source.
